tests/common.hpp

```diff
@@ -22,6 +22,57 @@
 
 #pragma once
 
+#include <map>
+#include <cuda.h>
+
+static std::map<CUdeviceptr, CUdeviceptr> _allocations;
+
+static inline CUresult alignedCUMemAlloc(CUdeviceptr *pptr, size_t psize, bool set_sync_memops)
```

Comment:
style note, this is C++ so how about having:
- a new API which support both regular and aligned allocations
- possibly in a C++ namespace, like gdrcopy::
- with a nifty though overloaded name, like gpuMemAlloc() :)

Reply:
Can you clarify a few points?
1. Are you proposing that we should introduce a new API to libgdrapi? Or would we confine this in unit tests only?
2. Regular and aligned allocations would use the same interface, i.e., gpuMemAlloc(CUdeviceptr *, size_t) -- that bool set_sync_memops is optional. Overloading will not work. Maybe we should go with flags instead? We can make that set_sync_memops as another flag.

Reply:
just an idea, gpuMemAlloc(CUdeviceptr *pptr, size_t psize, bool set_sync_memops = true, bool align_to_gpu_page = true).
that would still be part of the internal test APIs

Reply:
probably wrapped in either anonymous or gdrcopy::test namespace